traitsui/testing/tester/_ui_tester_registry/qt4/_traitsui/tree_editor.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+from traitsui.qt4.tree_editor import SimpleEditor
+
+
+from traitsui.testing.tester.command import (
+    MouseClick, MouseDClick, KeyClick, KeySequence
+)
+from traitsui.testing.tester.locator import TreeNode
+from traitsui.testing.tester.query import DisplayedText
+from traitsui.testing.tester._ui_tester_registry.qt4 import (
+    _interaction_helpers
+)
+
+from traitsui.testing.tester._ui_tester_registry._common_ui_targets import (
+    BaseSourceWithLocation
+)
+from traitsui.testing.tester._ui_tester_registry._traitsui_ui import (
+    register_traitsui_ui_solvers,
+)
+
+class _SimpleEditorWithTreeNode(BaseSourceWithLocation):
+    source_class = SimpleEditor
+    locator_class = TreeNode
+    handlers = [
+        (MouseClick, lambda wrapper, _: wrapper._target._mouse_click(
+            delay=wrapper.delay)),
+        (MouseDClick, lambda wrapper, _: wrapper._target._mouse_dclick(
+            delay=wrapper.delay)),
+        (KeySequence,
+            lambda wrapper, action: wrapper._target._key_sequence(
+                sequence=action.sequence,
+                delay=wrapper.delay,
+            )),
+        (KeyClick,
+            lambda wrapper, action: wrapper._target._key_press(
+                key=action.key,
+                delay=wrapper.delay,
+            )),
+        (DisplayedText,
+            lambda wrapper, _:  wrapper._target._get_displayed_text()),
+    ]
+
+    @classmethod
+    def register(cls, registry):
+        """ Class method to register interactions on a
+        _SimpleEditorWithTreeNode for the given registry.
+
+        If there are any conflicts, an error will occur.
+
+        Parameters
+        ----------
+        registry : TargetRegistry
+            The registry being registered to.
+        """
+        super().register(registry)
+        register_traitsui_ui_solvers(
+            registry=registry,
+            target_class=cls,
+            traitsui_ui_getter=lambda target: target._get_nested_ui()
+        )
+
+    def _get_model_view_index(self):
+        tree_widget = self.source._tree
+        i_column = self.location.column
+        i_rows = iter(self.location.row)
+        item = tree_widget.topLevelItem(next(i_rows))
+        for i_row in i_rows:
+            item = item.child(i_row)
+        q_model_index = tree_widget.indexFromItem(item, i_column)
+        return dict(
+            model=tree_widget.model(),
+            view=tree_widget,
+            index=q_model_index,
+        )
```

Comment:
This logic and the `TreeNode` object having a `row`/`column` I need to think about more carefully.  I had this code in a branch locally from a long time ago which I believe had originally been pulled from one of Kit's draft PRs.

There may be a simple way to do this / at the very least I need to better document the TreeNode class to say what row and column actually specify.  In this case (see test_TreeEditor_demo.py) row is a tuple containing the index of the node of interest at subsequent levels of the tree, and column is the index at that last level (?).  I have just been using column as 0, ... it may be possible we only need one single tuple to do this TreeNode location.